### ISSUE TITLE
RequestHandler bug

### ISSUE THREAD

**Issue by llonchj** [gocolly/colly#237]
I found a bug setting a `Collector.RequestHandler` func after a collector is initialized.

Setting Collector.RequestHandler, requires to execute `c.backend.Client.CheckRedirect=c.checkRedirectFunc()`. As `c.backend` is not accessible outside colly package, i will suggest to introduce a breaking change (might require version update according to golang/mod best practices) by renaming `Collector.RedirectHandler` to `Collector.redirectHandler`.

```go
// SetRedirectHandler instructs the Collector to allow multiple downloads of the same URL
func (c *Collector) SetRedirectHandler(f func(req *http.Request, via []*http.Request) error) {
	c.redirectHandler = f
	c.backend.Client.CheckRedirect = c.checkRedirectFunc()
}
```

I can make a PR to address the issue. @asciimoo any thoughts?

**Comment by asciimoo** [comment]
I'm not sure if I understand the problem. `c.backend.Client.CheckRedirect` doesn't need to be reassigned if the `c.RedirectHandler`changes because `c.RedirectHandler` is executed in the function returned by `c.checkRedirectFunc()` (https://github.com/gocolly/colly/blob/master/colly.go#L1133).
Could you provide a testcase which reproduces the bug?

**Comment by llonchj** [comment]
[Here](https://gist.github.com/llonchj/da8b782c06c7ca45744e3af66753133b) @asciimoo!

**Comment by asciimoo** [comment]
@llonchj you're right again, my pointer magic doesn't work here. Your solution seems to be our best option. Could you send me a PR with the changes?

**Comment by llonchj** [comment]
Ready @asciimoo

**Comment by StantonMatt** [comment]
This looks covered by the merged fix in #295.

Current `master` has `Collector.SetRedirectHandler(...)`, which stores the new handler and refreshes `c.backend.Client.CheckRedirect`. I checked the current path with a temporary local test that creates a collector, calls `SetRedirectHandler` after initialization, then visits a redirecting `httptest` server. The handler error is returned from `Visit`, so the updated handler is being used by the HTTP client.

Verification on current `master`:

```text
go test . -run 'TestIssue237SetRedirectHandlerAfterInit|Redirect|SetRedirectHandler' -count=1 -v
git diff --check origin/master...HEAD
```

Those passed locally.
